packages/compiler/src/core/checker.ts

```diff
@@ -2737,6 +2738,29 @@ export function createChecker(program: Program): Checker {
         }
       }
 
+      function checkForSelfReference(node: ModelSpreadPropertyNode, parent: ModelStatementNode) {
+        let targetName = "";
```

Comment:
This I think won’t work if you have an alias in between. I think you need to check the symbol resolved

Reply:
Updated!